### ISSUE TITLE
Header folding broken with ```fenced``` blocks (the whole document is folded)

### ISSUE THREAD

**Issue by eugenesvk** [SublimeText-Markdown/MarkdownEditing#733]
broken MarkdownEditing is the only plugin installed in safe mode

https://github.com/SublimeText-Markdown/MarkdownEditing/assets/12956286/82ff41f9-917a-48e5-9107-4270b6b4d1e7

correct (no plugins installed, safe mode)

https://github.com/SublimeText-Markdown/MarkdownEditing/assets/12956286/0a91c9f9-1d4f-4b07-8137-58cdc8d9e6f6

**Comment by deathaxe** [comment]
MarkdownEditing syntax and folding rules from https://github.com/sublimehq/Packages with few extensions to support more syntaxes in fenced code blocks.

The issue is caused by introducing syntax based folding for fenced code-blocks in https://github.com/sublimehq/Packages/pull/3782 and yet another example for how "reliable" ST's syntax based folding works in "expected" ways.

That's mainly another example for https://github.com/sublimehq/sublime_text/issues/6065.

As soon as a foldBegin or foldEnd selectors match multiple tokens, ST creates unbalanced sets of folding markers, causing content to fold in unexpected and hard to follow ways.

Needs some investigation to find a solution.
